src/charm.py

```diff
@@ -189,12 +186,22 @@ def __init__(self, *args):
             refresh_event=self.on.grafana_pebble_ready,  # pyright: ignore
         )
 
+        # -- database relation
+        self._db_name = f"{self._topology.application}-grafana-k8s-{self._topology.model_uuid}"
+        if self.model.relations[PGSQL_RELATION]:
+            self._db = DatabaseRequires(self, relation_name=PGSQL_RELATION, database_name=self._db_name)
+            self._db_type = "postgres"
+        else:
+            self._db = None
+            self._db_type = "sqlite3"
```

Comment:
Minor change: remove unnecessary `else`

```suggestion
        self._db = None
        self._db_type = "sqlite3"

        if self.model.relations[PGSQL_RELATION]:
            self._db = DatabaseRequires(self, relation_name=PGSQL_RELATION, database_name=self._db_name)
            self._db_type = "postgres"
```